ydb/core/kqp/host/kqp_gateway_proxy.cpp

```diff
@@ -792,7 +793,86 @@ class TKqpGatewayProxy : public IKikimrGateway {
     TFuture<TGenericResult> ModifyPermissions(const TString& cluster,
         const TModifyPermissionsSettings& settings) override
     {
-        FORWARD_ENSURE_NO_PREPARE(ModifyPermissions, cluster, settings);
+        CHECK_PREPARED_DDL(ModifyPermissions);
+
+        auto modifyPermissionsPromise = NewPromise<TGenericResult>();
```

Comment:
As far as  I understand, all this new code is for case if `IsPrepare()`. Then it will be better to move it under `if (IsPrepare())`